Library/Homebrew/test/test_language_python.rb

```diff
@@ -71,18 +73,27 @@ def test_pip_install_accepts_resource
     @venv.pip_install res
   end
 
-  def test_link_scripts_links_scripts
-    bin = (@dir/"bin")
-    dest = (@dir/"dest")
+  def test_pip_install_and_link_links_scripts
+    bin = @dir/"bin"
     bin.mkpath
-    refute((bin/"kilroy").exist?)
-    refute((dest/"kilroy").exist?)
+    dest = @formula.bin
+
+    refute_predicate bin/"kilroy", :exist?
+    refute_predicate dest/"kilroy", :exist?
+
     FileUtils.touch bin/"irrelevant"
-    @venv.link_scripts(dest) { FileUtils.touch bin/"kilroy" }
-    assert((bin/"kilroy").exist?)
-    assert((dest/"kilroy").exist?)
-    assert((dest/"kilroy").symlink?)
+    bin_before = Dir[bin/"*"]
+    FileUtils.touch bin/"kilroy"
+    bin_after = Dir[bin/"*"]
+    @venv.expects(:pip_install).with { |*params| params == ["foo"] }
```

Comment:
You can just say `.with("foo")`: http://gofreerange.com/mocha/docs/Mocha/Expectation.html#with-instance_method